engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -86,80 +94,125 @@ class EngineJobExecutionActor(replyTo: ActorRef,
+      if (effectiveCallCachingMode.readFromCache) {
+        factory.cacheHitCopyingActorProps(jobDescriptor, initializationData, serviceRegistryActor) match {
+          case Some(props) =>
+            callCachingActorProps = props
+          case None =>
+            log.warning("{}: Turning off cache reading for this job as no cache copying actor is available for this backend.", jobTag)
+            effectiveCallCachingMode = effectiveCallCachingMode.withoutRead
+            writeToMetadata(Map(effectiveCallCachingKey -> effectiveCallCachingMode.toString))
+        }
+      }
       effectiveCallCachingMode match {
         case activity: CallCachingActivity if activity.readFromCache =>
           initializeJobHashing(jobDescriptor, activity)
-          goto(CheckingCallCache) using JobDescriptorData(jobDescriptor, bjeaProps)
+          goto(CheckingCallCache) using updatedData
         case activity: CallCachingActivity =>
           initializeJobHashing(jobDescriptor, activity)
-          runJob(jobDescriptor, bjeaProps)
-        case CallCachingOff => runJob(jobDescriptor, bjeaProps)
+          runJob(updatedData)
+        case CallCachingOff => runJob(updatedData)
       }
     case Event(response: BackendJobPreparationFailed, NoData) =>
       replyTo forward response
       context stop self
       stay()
   }
 
-  // When CheckingCallCache, the FSM always has EJEAJobDescriptorData
   private val callCachingReadResultMetadataKey = "Call caching read result"
   when(CheckingCallCache) {
-    case Event(HashError(t), JobDescriptorData(jobDescriptor, bjeaProps)) =>
-      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Hashing Error: ${t.getMessage}"))
-      recordHashError(t)
-      runJob(jobDescriptor, bjeaProps)
-    case Event(CacheMiss, JobDescriptorData(jobDescriptor, bjeaProps)) =>
+    case Event(CacheMiss, data: ResponsePendingData) =>
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
-      log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
-      runJob(jobDescriptor, bjeaProps)
-    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, _)) =>
-      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID $cacheResultId)"))
-      log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
-      lookupCachedResult(jobDescriptor, cacheResultId)
+      log.debug("Cache miss for job {}", jobTag)
+      runJob(data)
+    case Event(hit: CacheHit, data: ResponsePendingData) =>
+      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID ${hit.cacheResultId})"))
+      log.debug("Cache hit for {}! Fetching cached result {}", jobTag, hit.cacheResultId)
+      fetchCachedResults(data, jobKey.call.task.outputs, hit)
+    case Event(HashError(t), data: ResponsePendingData) =>
```

Comment:
I had a look at what happens if there's a CCRID lookup error: the EJHA throws a runtime exception. Presumably if that happens, the entire EJEA also fails?
Is that appropriately consistent with `HashError` which merely disables call caching and carries on?